wgpu-hal/src/gles/adapter.rs

```diff
@@ -184,8 +191,18 @@ impl super::Adapter {
     ) -> Option<crate::ExposedAdapter<super::Api>> {
         let gl = context.lock();
         let extensions = gl.supported_extensions();
-
-        let (vendor_const, renderer_const) = if extensions.contains("WEBGL_debug_renderer_info") {
+        // if emscripten, has debug extension and if we can enable it -> use unmasked vendor/renderer
+        // https://github.com/gfx-rs/wgpu/issues/3245
+        let (vendor_const, renderer_const) = if cfg!(target_os = "emscripten")
+            && extensions.contains("WEBGL_debug_renderer_info")
+            && unsafe {
+                emscripten_webgl_enable_extension(
+                    emscripten_webgl_get_current_context(),
+                    "WEBGL_debug_renderer_info\0".as_ptr() as _,
+                )
+            } == 1
+        {
```

Comment:
The logic isn't right here.

It should be:

- if extension:
  - if emscripten:
    - activate extension
  - use extension

We don't need to check if activation fails, as it shouldn't fail

Reply:
what if it fails though? better to fallback to vendor/renderer instead of crashing.

Reply:
Fair enough

Reply:
The logic still needs to be changed here to what I listed, just with an extra conditional at the end

Reply:
The logic as it stands, excludes regular webgl wasm

Reply:
oh! you are right. I was looking at the web.rs file in the same directory and smh assumed that all w-u-u logic is separate from gles. thanks for pointing this out

Reply:
done. and tested on ff/chrome linux.